index.js

```diff
@@ -127,15 +140,13 @@ KinesisStream.prototype._sendEntries = function () {
 };
 
 KinesisStream.prototype._write = function (chunk, encoding, done) {
-  var self = this;
-
   if (!this._params.streamName) {
-    return immediate (done, new Error('Stream\'s name was not set.'));
+    return setImmediate (done, new Error('Stream\'s name was not set.'));
   }
 
   try {
     var msg = chunk.toString();
```

Comment:
I think that it could be either of both. Maybe the line must be written like this:
`var msg = chunk.toString(encoding);`